### DISCUSSION TITLE
docker-compose up not working if tty-connection not up

### DISCUSSION THREAD

**Discussion by EarlOfBurton** [OctoPrint/docker#199] [Q&A]
Hello,
I'm new to docker and containers but pretty happy, thanks to your guides, to get two printers running!

I noticed that when I am "starting" my two octoprint instances via "docker-compose up" they will not start, if there are not active ttyUSB0 (printer 1) and ttyACM0 (printer 2)-connections. They will error and not run.

Is it somehow possible to start anyway (I don't have both printers online all the time)?

The exact error is:

`pi@dockerpi:~ $ docker restart octoprint-docker-master_octoprint1_e3v2_1 
Error response from daemon: Cannot restart container octoprint-docker-master_octoprint1_e3v2_1: linux runtime spec devices: error gathering device information while adding custom device "/dev/ttyE3V2": no such file or directory
`

Thanks for your help to a newbie in advance.

**Comment by FirePanther** [comment]
Hey, probably the OP won't see this but this page was highly ranked on google and I thought if someone should have a similar problem, they can find an idea how to fix it.
The way I fixed it is, that I, when starting Octoprint, I turn on the 3D printer for a few seconds. The problem why this happens is only, because Docker needs the device once and you can't ignore the error. That's why I fixed it this way:

```version: '3.8'

services:
  octoprint-init:
    image: curlimages/curl
    command: curl -X POST https://your-homeserver/start3dPrinter
    restart: "no"

  octoprint:
    image: octoprint/octoprint:latest
    container_name: octoprint
    depends_on:
      octoprint-init:
        condition: service_completed_successfully
    restart: unless-stopped
    volumes:
      - ./config:/octoprint
    devices:
      - /dev/serial/by-id/usb-...-port0:/dev/ttyUSB0
    environment:
      - ENABLE_MJPG_STREAMER=true
      - OCTOPRINT_SERVER_PORT=5000
    network_mode: host```

The homeserver is just a node server which checks, if the printer is currently turned on (if so, don't do anything), if it is not turned on, turn it on (wait 5 seconds, just to be sure the device is connected to your machine), respond to the request, (wait again a little bit, so docker can start the container) and turn it off again
some pseudo code:
```if (!printerTurnedOn) {
  turnOnPrinter()
  setTimeout(_ => {
    respond('ok')
    setTimeout(_ => {
      turnOffPrinter()
    }, 1000)
  }, 5000)
} else {
  respond('ok')
}```
(The way how I am in general turning on and off my printer is with Home Assistant and smart outlets.)
